packages/schematics/angular/migrations/update-14/angular-packages-version-prefix.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { Rule } from '@angular-devkit/schematics';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import { NodeDependencyType } from '../../utility/dependencies';
+import { JSONFile } from '../../utility/json-file';
+
+const PACKAGES_REGEXP = /^@(?:angular|nguniversal|schematics|angular-devkit)\/|^ng-packagr$/;
+
+export default function (): Rule {
+  return (tree, context) => {
+    const json = new JSONFile(tree, '/package.json');
+    updateVersionPrefixToTilde(json, NodeDependencyType.Default);
+    updateVersionPrefixToTilde(json, NodeDependencyType.Dev);
```

Comment:
Are there any (maybe lesser-used) packages that would be placed in peer deps or optional deps? I just want to make sure there's no need to migrate other forms of dependencies as well. Would it be safer to update those too?

Reply:
Not in the root of the package.json. 

There might be peer deps in the `package.json` of library projects. But these are generally added by users to specify which versions of Angular their library supports, so I don't think we should amend those.

Reply:
Sounds good, just wanted to verify.